### ISSUE TITLE
Can Card.IO be made any smaller please?

### ISSUE THREAD

**Issue by rsaunders100** [card-io/card.io-iOS-SDK#177]
Assuming:
- This library saves ~10s over manual entry.
- The average user spends an extra ~2.5 sec downloading the app.  

Then if less than 1 in 4 people use the scan feature, then it's not worth it.

I can see you have already done a huge amount by squeezing openCV down, but It it possible to do anymore? Unused code? Unused assets? Supporting app slicing?

**Comment by josharian** [comment]
There are really two size issues, pre- and post-linking.

The former was made much, much worse recently (see e.g. #147, #155, and #167, among others), and there's probably a way get it back to how it was before bitcode ruined the party. However, the pre-linking size doesn't impact the user download time.

As for the post-linking size, there has been some discussion at https://github.com/card-io/card.io-iOS-source/issues/27. I fear that there are no significant, obvious wins available.

Contributions on either front would be most welcome. The pre-linking size can probably just be fixed, but I don't think that's what you're worried about. The post-linking size will almost certainly involve trade-offs. I strongly recommend discussing any proposed trade-offs before starting work on them.

**Comment by younanjo** [comment]
My app has gone from 2MB to 302MB with card.io framework added! Is there any workarounds for this?

**Comment by bluk** [comment]
@younanjo Be sure that you're following how the sample project is being built. I have had a couple of people email me recently where they were copying the static libraries into the application instead of just linking to it.

**Comment by QoT** [comment]
Source code really is* 300 MB larger, but a whole application's build (IPA file) is only an about 30 MB in total.
